src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,37 @@
-import { createSlice } from '@reduxjs/toolkit';
-import { v4 as uuidv4 } from 'uuid';
+import { createAsyncThunk } from '@reduxjs/toolkit';
 
-const initialState = [
-  {
-    id: uuidv4(),
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    id: uuidv4(),
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    id: uuidv4(),
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
+const GET_BOOKS = 'GET_BOOKS';
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Hh517pOPUdJbRWvGSr6d/books/';
 
-const bookSlice = createSlice({
-  name: 'bookstore/books',
-  initialState,
-  reducers: {
-    addBook: (state, action) => {
-      const newBook = {
-        id: uuidv4(),
-        title: action.payload.title,
-        author: action.payload.author,
-      };
-      state.push(newBook);
-    },
-    removeBook: (state, action) => state.filter((book) => book.id !== action.payload.id),
-  },
+export const getAllBooks = createAsyncThunk(GET_BOOKS, async (_, { dispatch }) => {
+  const response = await fetch(url);
+  const data = await response.json();
+  const books = Object.keys(data).map((id) => ({ ...data[id][0], item_id: id }));
+  dispatch({ type: GET_BOOKS, payload: books });
 });
 
-export const { addBook, removeBook } = bookSlice.actions;
-export default bookSlice.reducer;
+export const addBook = createAsyncThunk(ADD, async (book, { dispatch }) => {
+  await fetch(url, { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(book) });
+  dispatch({ type: ADD, payload: book });
+});
+
+export const removeBook = createAsyncThunk(REMOVE, async (id, { dispatch }) => {
+  await fetch(`${url}/${id}`, { method: 'DELETE' });
+  dispatch({ type: REMOVE, payload: id });
+});
+
```

Comment:
- [ ] Please make sure that the functionality of the applications is working fine. While I was testing your application as you can see from the video below the `add` and `remove` function to the Bookstore API is not working perfectly as expected. When I add the book and refresh the page I can't see the book there which means the book is not added to the bookstore API. Also when I click on the remove button, the book is not been removed. :relaxed: So please make sure that you fix it. :pray: 

[Screencast from 03-01-2023 01:30:06 PM.webm](https://user-images.githubusercontent.com/92303891/222155062-9e4bc1cd-0162-4c34-91a0-148278ebc097.webm)